cli-toolchain/cognito-triggers.md

```diff
@@ -0,0 +1,508 @@
+
+app.component.ts:
+```js
+import { Component } from '@angular/core';
+import { AmplifyService } from 'aws-amplify-angular';
+import Auth from '@aws-amplify/auth';
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent {
+  title = 'cli-lambda-sample';
+  confirmSignUp: boolean;
+  user: any;
+  constructor( private amplifyService: AmplifyService ) {
+    this.amplifyService.authStateChange$
+        .subscribe(authState => {
+            this.confirmSignUp = authState.state === 'confirmSignUp';
+            if (!authState.user) {
+                this.user = null;
+            } else {
+                this.user = authState.user;
+            }
+    });
+  }
+}
+```
+
+app.component.html:
+```html
+<amplify-authenticator [hide]="['ConfirmSignUp']"></amplify-authenticator>
+<div  *ngIf="confirmSignUp">
+  Check your email account for a confirmation message!
+</div>
+
+```
+
+#### Vue Sample
+
+The following is an example of how to configure the aws-amplify-vue authenticator components so that your app displays a message telling the user to check their email, instead of showing the default 'ConfirmSignUp' component.
+
+```javascript
+<template>
+  <div id="app">
+    <amplify-sign-up v-if="signUp"></amplify-sign-up>
+    <amplify-sign-in v-if="signIn"  v-bind:usernameAttributes="usernameAttributes"></amplify-sign-in>
+    <div v-if="confirming">Check your email for a confirmation email</div>
+    <amplify-sign-out
+      v-if="signedIn"
+    ></amplify-sign-out>
+  </div>
+</template>
+
+<script>
+import { components } from 'aws-amplify-vue'
+import { AmplifyEventBus } from 'aws-amplify-vue';
+import VueRecaptcha from 'vue-recaptcha';
+
+
+export default {
+  name: 'app', 
+  components: {
+    VueRecaptcha,
+    ...components
+  },
+  data() {
+    return {
+      signIn: true,
+      signUp: false,
+      confirming: false,
+      usernameAttributes: 'username',
+      user: {},
+    }
+  },
+  mounted: async function () {
+    await this.$Amplify.Auth.currentAuthenticatedUser()
+      .then((user) => {
+        this.user = user;
+        this.signIn = false;
+      })
+      .catch((e) => {
+        console.log('currentUser error', e)
+      })
+    AmplifyEventBus.$on('authState', info => {
+      this.signIn = info === 'signIn' || info === 'signedOut';
+      this.confirming = info === 'confirmSignUp';
+      this.signUp = info === 'signUp';
+      this.signedIn = info === 'signedIn';
+    });
+    AmplifyEventBus.$on('localUser', user => {
+      this.user = user;
+    });
+  }
+}
+</script>
+```
+
+### Add User to Group
+
+This trigger allows you to define a Cognito group to which a user will be added upon registration.  
+
+Please note that you will need to create the Cognito group manually.
```

Comment:
I don't think this is true any longer with Dustin's latest build.

Reply:
Replacing with this:

> The trigger will check for the existence of the group in your User Pool, and will create the group if it is not present.